### PR TITLE
Add classpath support for git hooks

### PR DESCRIPTION
**Pull request by nhojpatrick**
This PR is to add support to retreive git hooks from a dependency.

So something like this could automatically pick up any new hooks and install them via a new `tld.example:company-git-hooks:1.x.x` release.

```
<plugin>
	<groupId>com.rudikershaw.gitbuildhook</groupId>
	<artifactId>git-build-hook-maven-plugin</artifactId>
	<version>3.1.0-SNAPSHOT</version>
	<configuration>
		<installHooks>
			<pre-commit>company-pre-commit-hook</pre-commit>
		</installHooks>
	</configuration>
	<dependencies>
		<dependency>
			<groupId>tld.example</groupId>
			<artifactId>company-git-hooks</artifactId>
			<version>[1.0.0,1.999.999)</version>
		</dependency>
	</dependencies>
</plugin>
    ```

Outstanding Tasks;
[x] Clean Up Code for copying
[x] Clean Up Code for setting exec bit if form classpath
[x] Unit Testing
[x] Check if commons-io is really need for Resource to File copying

### CODE REVIEW

**Comment by nhojpatrick** [comment]
I think this is ready for review now, i did create new tests but instead update the existing install-hook and reinstall-hook tests.

**Comment by rudikershaw** [comment]
Thanks very much. I'm hoping to find some time over the next couple of days where I can sit down and give this the time it deserves. Will keep you posted.

File: README.md [hunk 1]
```diff
@@ -34,6 +34,13 @@ A common use-case might be to install local git hooks by setting the `core.hooks
           <custom.configuration>true</custom.configuration> 
         </gitConfig>
       </configuration>
+      <dependencies>
```
Comment (rudikershaw) [inline README.md:37]: This is in the wrong section in the readme. This example won't do anything as it is only setting the `core.hooksPath` to point at a directory. You probably wanted this in the second example where you have added `class_path/package/hook.sh`.

**Review by rudikershaw — CHANGES_REQUESTED** [review]
Hi there. Apologies for the delay. I have a few minor suggestions.

File: pom.xml [hunk 2]
```diff
@@ -143,6 +148,19 @@
               <goal>jar</goal>
             </goals>
           </execution>
+          <execution>
+            <id>package-cphooks-for-testing</id>
+            <phase>process-test-resources</phase>
+            <configuration>
+              <classifier>CPHOOKS</classifier>
+              <includes>
+                <include>com/rudikershaw/gitbuildhook/tests/cphooks/**</include>
```
[the remark this answers is not in the document]
Comment (nhojpatrick) [reply pom.xml:157]: If I move the testing files underneath `src/main/resources` then I would be able to use the current `execution` block defined as lines `141-150`. But I didn't want to ship these test files with the final jar.

The current block creates a jar with the classifier of `TEST`, but it still the contents from `src/main`.
I've tried to see if I can update it to do basically both a `jar` and `test-jar` but have not been able to.

If you have an idea how I can change lines `141-150` so it still has content from `src/main` and also `src/test`, I'll try again.

File: README.md (hunk 1 above)
Comment (nhojpatrick) [reply README.md]: Have moved to the correct section.

File: pom.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (nhojpatrick) [reply pom.xml:157]: My only idea to just have on `execution` is to add the testing hooks under `src/main/resources` then add an excludes to the default jar `package` to excluded them from the released jar. But feels a bit dodgy to me, doing that.

**Review by rudikershaw — APPROVED** [review]

**Comment by rudikershaw** [comment]
Thanks again for your contribution. There are a few more things I want to attend to in this release, but after those are sorted I will see to getting this released in 3.1.0.

**Pull request merged**